src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -71,7 +71,7 @@
      *             if serverAddresses is empty or resolves to an empty list
      */
     public StaticHostProvider(Collection<InetSocketAddress> serverAddresses) {
-       sourceOfRandomness = new Random(System.currentTimeMillis() ^ this.hashCode());
+       sourceOfRandomness = new Random(System.nanoTime() ^ this.hashCode());
```

Comment:
@asdf2014 
- shift right a space(haha,original code is also not alignment)
- according to [ZOOKEEPER-1990](https://issues.apache.org/jira/browse/ZOOKEEPER-1990)
   >    the goal here is to make sure that when multiple client threads start at the same time on the same machine 
         (such as in our systest) and execute this line each one is likely to have a different random seed.
- your demo is not convincing because different client have different values of `this.hashCode()`(different instances) although the elements in inetSocketAddressesList is same. Look at my [demo](https://github.com/maoling/fuck_zookeeper/blob/master/src/java/test/com/apache/zookeeper/self/ZOOKEEPER_2840.java)

Reply:
Hi, @maoling. Sorry for my belated comment. Indeed, i hadn't been considered this case that different clients will own different `hashcode`s. Because, in my sistuation, our Zookeeper clusters only give service to some apointed BigData components, e.g., [HBase](https://yuzhouwan.com/posts/45888/), [Druid.io](https://yuzhouwan.com/posts/5845/), [Hadoop](https://yuzhouwan.com/posts/60504/), [Kafka](https://yuzhouwan.com/posts/26002/) etc. Inside these components will have settled clients, so them can not offer too many different `hashcode`s.